docker/go-control-plane/controlplane/resource.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Envoyproxy Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package controlplane
+
+import (
+	"strconv"
+	"time"
+
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+
+	pb_cluster "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	pb_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	pb_endpoint "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	pb_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	pb_route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	pb_router "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/router/v3"
+	pb_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+)
+
+const (
+	ClusterName  = "example_proxy_cluster"
+	ListenerName = "listener_0"
+)
+
+var snapshot_version int = 1;
+
+// MakeCluster builds a CDS resource with a given clusterName that points
+// the users to upstreamHost:upstreamPort
+func MakeCluster(clusterName, upstreamHost string, upstreamPort uint32) *pb_cluster.Cluster {
```

Comment:
At this point, this package exports two functions which are basically helpers to create xDS resources. This would well be in a separate file in the `main` package and then it would not have to be exported either.

Reply:
This functionality will move to the Python. Would like to keep it separately here as this is the last file from the Envoy code.